src/coins.cpp

```diff
@@ -398,11 +400,19 @@ bool CCoinsViewCache::HavePourRequirements(const CTransaction& tx) const
         }
 
         ZCIncrementalMerkleTree tree;
-        if (!GetAnchorAt(pour.anchor, tree)) {
-            // If we do not have the anchor for the pour,
-            // it is invalid.
+        auto it = intermediates.find(pour.anchor);
+        if (it != intermediates.end()) {
+            tree = it->second;
+        } else if (!GetAnchorAt(pour.anchor, tree)) {
             return false;
         }
+
+        BOOST_FOREACH(const uint256& commitment, pour.commitments)
+        {
+            tree.append(commitment);
+        }
+
+        intermediates.insert(std::make_pair(tree.root(), tree));
```

Comment:
The RPC will need to construct the joinsplits in such a way that the commitments it uses as "change" for previous joinsplits are available to the newer ones. It can indeed do so.